src/Microsoft.AspNet.StaticFiles/LoggerExtensions.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.StaticFiles
+{
+    /// <summary>
+    /// Defines *all* the logger messages produced by static files
+    /// </summary>
+    internal static class LoggerExtensions
+    {
+        private static Action<ILogger, string, Exception> _logMethodNotSupported;
+        private static Action<ILogger, string, string, Exception> _logFileServed;
+        private static Action<ILogger, string, Exception> _logPathMismatch;
+        private static Action<ILogger, string, Exception> _logFileTypeNotSupported;
+        private static Action<ILogger, string, Exception> _logFileNotFound;
+        private static Action<ILogger, string, Exception> _logPathNotModified;
+        private static Action<ILogger, string, Exception> _logPreconditionFailed;
+        private static Action<ILogger, string, Exception> _logNoPrecondition;
+        private static Action<ILogger, int, string, Exception> _logHandled;
+        private static Action<ILogger, string, Exception> _logRangeNotSatisfiable;
+        private static Action<ILogger, StringValues, string, Exception> _logSendingFileRange;
+        private static Action<ILogger, StringValues, string, Exception> _logCopyingFileRange;
+        private static Action<ILogger, long, string, string, Exception> _logCopyingBytesToResponse;
+        private static Action<ILogger, string, Exception> _logMultipleFileRanges;
+
+        static LoggerExtensions()
+        {
+            _logMethodNotSupported = LoggerMessage.Define<string>(
+                logLevel: LogLevel.Debug,
```

Comment:
I talked to Ryan and he said that most of these should be Verbose, so you always end up with at least one verbose message when the middleware does not handle a request. If it does handle the request then there should be one or more Informational messages.